src/java_bytecode/java_string_library_preprocess.cpp

```diff
@@ -1723,6 +1723,94 @@ codet java_string_library_preprocesst::make_string_to_char_array_code(
 
 /*******************************************************************\
 
+Function: java_string_library_preprocesst::make_object_get_class_code
+
+  Inputs:
+    type - type of the function called
+    loc - location in the source
+    symbol_table - the symbol table
+
+  Outputs: Code corresponding to
+          > Class class1;
+          > string_expr1 = substr(this->@class_identifier, 6)
```

Comment:
This is to remove the "java::" prefix. (see the comment on line 1778 https://github.com/diffblue/cbmc/pull/945/files/556dc9a262b23686d07d9019f5d5f69bf8fee5ab#diff-333608a53b33706cdc28c1f3c1d0ca16R1778)
Maybe `std::string("java::").length()` would be more explicit...